File: app/src/main/java/com/annapol04/munchkin/db/ScopeTypeConverter.java
package com.annapol04.munchkin.db;

import android.arch.persistence.room.TypeConverter;

import com.annapol04.munchkin.engine.Scope;

public class ScopeTypeConverter {
    @TypeConverter
   public static int toInt(Scope scope) {
        return scope.ordinal();
    }

    @TypeConverter
    public static Scope toScope(int id) {
        return Scope.Companion.fromId(id);
    }
}

File: app/src/main/java/com/annapol04/munchkin/db/ScopeTypeConverter.kt
package com.annapol04.munchkin.db

import android.arch.persistence.room.TypeConverter

import com.annapol04.munchkin.engine.Scope

class ScopeTypeConverter {
    @TypeConverter
    fun toInt(scope: Scope): Int {
        return scope.ordinal
    }

    @TypeConverter
    fun toScope(id: Int): Scope {
        return Scope.fromId(id)
    }
}
